src/api/app/controllers/webui/project_controller.rb

```diff
@@ -526,6 +526,8 @@ def load_project_info
 
     reqs = @project.open_requests
     @requests = (reqs[:reviews] + reqs[:targets] + reqs[:incidents] + reqs[:maintenance_release]).sort!.uniq
+    @incoming_requests = RequestsFinder.new.incoming(reqs.values.sum, @project.name)
+    @outgoing_requests = RequestsFinder.new.outgoing(reqs.values.sum, @project.name)
```

Comment:
You instantiate possibly very man requests and load them into memory in a instance variable....